domain/Lockset.go

```diff
@@ -2,40 +2,33 @@ package domain
 
 import (
 	"go/token"
+
 	"golang.org/x/tools/go/ssa"
 )
+
 type locksLasUse map[token.Pos]*ssa.CallCommon
 
 type Lockset struct {
 	ExistingLocks   locksLasUse
 	ExistingUnlocks locksLasUse
 }
 
-
 func NewLockset() *Lockset {
 	return &Lockset{
-		ExistingLocks:   make(locksLasUse, 0),
-		ExistingUnlocks: make(locksLasUse, 0),
+		ExistingLocks:   make(locksLasUse),
+		ExistingUnlocks: make(locksLasUse),
 	}
 }
 
 func (ls *Lockset) UpdateLockSet(newLocks, newUnlocks locksLasUse) {
-	if newLocks != nil {
-		for lockName, lock := range newLocks {
-			ls.ExistingLocks[lockName] = lock
-		}
-	}
 	for unlockName, _ := range newUnlocks {
-		if _, ok := ls.ExistingLocks[unlockName]; ok {
```

Comment:
Unlocks are never removed in your change so I would keep the first block (but still trimming that nil check). 
The algorithm works by remembering each lock's state (locked/unlocked/or nothing, of course). It means that if a mutex was unlocked at some point but later was locked again, then its latest status is locked, and the unlock status is removed. This is the formula. I hope it's more clear:

![image](https://user-images.githubusercontent.com/34520943/96378292-a3266800-1193-11eb-8036-801da2d6bdf2.png)

Reply:
@ninedraft idea was about 'no need to check if element is in map before deletion'. And that make sense, 'cause everything from `newUnlocks` will be removed anyway, isn't it?

Reply:
@cristaloleg what do you mean by "cause everything from `newUnlocks` will be removed anyway"?

Reply:
oh yeah, got it, sorry :)

Reply:
✅